code/game/objects/items/weapons/defib.dm

```diff
@@ -294,26 +279,67 @@
 	w_class = WEIGHT_CLASS_BULKY
 	resistance_flags = INDESTRUCTIBLE
 	toolspeed = 1
-
-	var/revivecost = 1000
-	var/cooldown = FALSE
-	var/busy = FALSE
 	base_icon_state = "defibpaddles"
+	/// Amount of power used on a shock.
+	var/revivecost = 1000
+	/// Active defib this is connected to.
 	var/obj/item/defibrillator/defib
+	/// Whether or not the paddles are on cooldown. Used for tracking icon states.
+	var/on_cooldown = FALSE
 
 /obj/item/twohanded/shockpaddles/New(mainunit)
-	..()
+	. = ..()
 	if(check_defib_exists(mainunit, src))
 		defib = mainunit
 		loc = defib
-		busy = FALSE
 		update_icon(UPDATE_ICON_STATE)
+		AddComponent(/datum/component/defib, actual_unit = defib, combat = defib.combat, safe_by_default = defib.safety, combat_defib_chance = defib.heart_attack_probability, emp_proof = defib.hardened, emag_proof = defib.emag_proof)
+	else
+		AddComponent(/datum/component/defib)
+	RegisterSignal(src, COMSIG_DEFIB_READY, PROC_REF(on_cooldown_expire))
+	RegisterSignal(src, COMSIG_DEFIB_SHOCK_APPLIED, PROC_REF(after_shock))
+	RegisterSignal(src, COMSIG_DEFIB_PADDLES_APPLIED, PROC_REF(on_application))
```

Comment:
On one hand, I think offloading a lot of behavior that could be on the component, to the paddles instead, hurts the component's usefulness, because you'll have to have do this for everything the component is applied to. But on the other hand, it's a defib component and probably won't be applied to much more than this.

Reply:
The only real behavior that ends up getting offloaded is handling cooldowns and charges, which I suppose could be handled by the component but would probably require a subclass of the component. These overrides (especially for things like the cooldown) certainly could be handled by the component, but I felt things like the recharge would be better off handled by each child as needed. It means there *could* be charge-deducting behavior in the component, but it would need to be handled for each different implementation anyway.

I could refactor it to move more behavior onto the component if that'd be desired.

Reply:
Yeah personally, I would like to see as much behavior on the component as possible.